File: client/src/app/modules/quotation/services/service.family.group.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { ServiceFamilyGroup } from '../../quotation/model/ServiceFamilyGroup';

@Injectable({
  providedIn: 'root'
})
export class ServiceFamilyGroupService extends AppRestService<ServiceFamilyGroup>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getServiceFamilyGroups() {
    return this.getList(new HttpParams(), "service-family-groups");
  }
  
   addOrUpdateServiceFamilyGroup(serviceFamilyGroup: ServiceFamilyGroup) {
    return this.addOrUpdate(new HttpParams(), "service-family-group", serviceFamilyGroup, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/quotation/service/ServiceFamilyGroupService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.ServiceFamilyGroup;

public interface ServiceFamilyGroupService {
    public List<ServiceFamilyGroup> getServiceFamilyGroups();

    public ServiceFamilyGroup getServiceFamilyGroup(Integer id);
	
	 public ServiceFamilyGroup addOrUpdateServiceFamilyGroup(ServiceFamilyGroup serviceFamilyGroup);
}
